flame/utils/mock_flame_core.py

```
@@ -44,11 +44,23 @@ def __init__(self, test_kwargs) -> None:
+
+    def increment(self):
+        self.iter += 1
+
+    def get_iterations(self):
+        return self.iter
+
+
 class MockFlameCoreSDK:
-    num_iterations: int = 0
+    num_iterations: IterationTracker = IterationTracker()
     logger: dict[str, list[str]] = {}
```

Ruff (0.15.6)

[warning] Mutable default value for class attribute

(RUF012)